src/main/java/at/hannibal2/skyhanni/features/mining/eventtracker/MiningEventTracker.kt

```diff
@@ -121,36 +134,54 @@ class MiningEventTracker {
             return
         }
 
-        val miningEventData = MiningEventData(
+        val miningEventData = MiningEventDataSend(
             LorenzUtils.skyBlockIsland,
             serverId,
             eventType,
             timeRemaining.inWholeMilliseconds,
             LorenzUtils.getPlayerUuid()
         )
         val miningEventJson = ConfigManager.gson.toJson(miningEventData)
-//         //todo remove
-//         println("\n```json$miningEventJson```")
         SkyHanniMod.coroutineScope.launch {
             sendData(miningEventJson)
         }
     }
 
-    private fun isEnabled() = (IslandType.DWARVEN_MINES.isInIsland() || IslandType.CRYSTAL_HOLLOWS.isInIsland())
-            && config.sendData
-//         && config.enabled
-
     private fun sendData(json: String) {
         val response = APIUtil.postJSON("https://api.soopy.dev/skyblock/chevents/set", json)
         if (!response.success) return
-        val success = response.data.getBoolean("success")
-        if (!success) {
-            val cause = response.data.getStringOrValue("cause", "unknown")
+
+        val formattedResponse = ConfigManager.gson.fromJson<MiningEventDataReceive>(response.data)
+        if (!formattedResponse.success) {
+            if (!config.enabled) return
             ErrorManager.logErrorWithData(
                 Exception("PostFailure"), "Sending mining event data was unsuccessful",
-                "cause" to cause,
+                "cause" to formattedResponse.cause,
                 "sentData" to json
             )
         }
     }
+
+    private fun fetchData() {
+        canRequestAt = SimpleTimeMark.now() + defaultCooldown
+        SkyHanniMod.coroutineScope.launch {
+            val data = APIUtil.getJSONResponse("https://api.soopy.dev/skyblock/chevents/get")
+            val miningEventData = ConfigManager.gson.fromJson(data, MiningEventDataReceive::class.java)
+
+            if (!miningEventData.success) {
+                ErrorManager.logErrorWithData(
+                    Exception("PostFailure"), "Sending mining event data was unsuccessful",
+                    "cause" to miningEventData.cause,
+                    "recievedData" to data
+                )
+                return@launch
+            }
+
+            canRequestAt = SimpleTimeMark.now() + miningEventData.data.updateIn.milliseconds
+
+            MiningEventDisplay.updateData(miningEventData.data)
+        }
+    }
+
+    private fun isEnabled() = (IslandType.DWARVEN_MINES.isInIsland() || IslandType.CRYSTAL_HOLLOWS.isInIsland())
```

Comment:
the name ```isEnabled``` seem to me a bit missleading for this function since it only checks the islands and nothing more.
Also isn't it simply ```LocationUtils.inAdvancedMiningIsland```

Reply:
You'd be right, I made the `LocationUtils.inAdvancedMiningIsland` after i made this